From 16d81453bd768ae4c879b5133fa91458c909f3e8 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Wed, 28 Dec 2022 01:01:37 +0800
Subject: [PATCH 12/28] Revert "drm: Fix potential null-ptr-deref in
 drm_vblank_destroy_worker()"

This reverts commit 1d160dfb3fdf11ba9447e862c548447f91f4e74a.
---
 drivers/gpu/drm/drm_internal.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/drm_internal.h b/drivers/gpu/drm/drm_internal.h
index 41efe40bc70f6..f80e0f28087d1 100644
--- a/drivers/gpu/drm/drm_internal.h
+++ b/drivers/gpu/drm/drm_internal.h
@@ -116,8 +116,7 @@ static inline void drm_vblank_flush_worker(struct drm_vblank_crtc *vblank)
 
 static inline void drm_vblank_destroy_worker(struct drm_vblank_crtc *vblank)
 {
-	if (vblank->worker)
-		kthread_destroy_worker(vblank->worker);
+	kthread_destroy_worker(vblank->worker);
 }
 
 int drm_vblank_worker_init(struct drm_vblank_crtc *vblank);
-- 
2.25.1

